JunList.java

```diff
@@ -1,70 +1,73 @@
 public class JunList {
-    private int[] list;
+    private Integer[] list;

```

Comment:
Why did you switch to Integer?

Reply:
so I could explicitly store null as a value in remove method. couldn't do that with int[]